<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Importing and preocessing data | Optimised human stool sample collection for multi-omic microbiota analysis R code</title>
  <meta name="description" content="Book containg the R code for the paper Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Importing and preocessing data | Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Book containg the R code for the paper Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Importing and preocessing data | Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  
  <meta name="twitter:description" content="Book containg the R code for the paper Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2024-07-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="01-Intro.html"/>
<link rel="next" href="03-Figure_3.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Optimised human stool sample collection for multi-omic microbiota analysis R code</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html"><i class="fa fa-check"></i><b>2</b> Importing and preocessing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html#qiime2r-import"><i class="fa fa-check"></i><b>2.1</b> qiime2R import</a></li>
<li class="chapter" data-level="2.2" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html#preprocess-data"><i class="fa fa-check"></i><b>2.2</b> Preprocess data</a></li>
<li class="chapter" data-level="2.3" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html#lipid-data"><i class="fa fa-check"></i><b>2.3</b> Lipid data</a></li>
</ul></li>
<li class="part"><span><b>Main figures</b></span></li>
<li class="chapter" data-level="3" data-path="03-Figure_3.html"><a href="03-Figure_3.html"><i class="fa fa-check"></i><b>3</b> Figure 3</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Figure_3.html"><a href="03-Figure_3.html#libraries"><i class="fa fa-check"></i><b>3.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-Figure_3.html"><a href="03-Figure_3.html#figure-3a"><i class="fa fa-check"></i><b>3.2</b> Figure 3A</a></li>
<li class="chapter" data-level="3.3" data-path="03-Figure_3.html"><a href="03-Figure_3.html#figure-3b"><i class="fa fa-check"></i><b>3.3</b> Figure 3B</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="03-Figure_3.html"><a href="03-Figure_3.html#percentage-of-bacteroides-and-firmicutes"><i class="fa fa-check"></i><b>3.3.1</b> Percentage of Bacteroides and Firmicutes</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="03-Figure_3.html"><a href="03-Figure_3.html#figure-3c"><i class="fa fa-check"></i><b>3.4</b> Figure 3C</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Figure_4.html"><a href="04-Figure_4.html"><i class="fa fa-check"></i><b>4</b> Figure 4</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Figure_4.html"><a href="04-Figure_4.html#libraries-1"><i class="fa fa-check"></i><b>4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2" data-path="04-Figure_4.html"><a href="04-Figure_4.html#genus"><i class="fa fa-check"></i><b>4.2</b> Genus</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-Figure_4.html"><a href="04-Figure_4.html#preprocess-table-and-convert-to-genus-table"><i class="fa fa-check"></i><b>4.2.1</b> Preprocess table and convert to genus table</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-Figure_4.html"><a href="04-Figure_4.html#maaslin2-on-genera"><i class="fa fa-check"></i><b>4.2.2</b> MaAsLin2 on genera</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Figure_5.html"><a href="05-Figure_5.html"><i class="fa fa-check"></i><b>5</b> Figure 5</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Figure_5.html"><a href="05-Figure_5.html#libraries-2"><i class="fa fa-check"></i><b>5.1</b> libraries</a></li>
<li class="chapter" data-level="5.2" data-path="05-Figure_5.html"><a href="05-Figure_5.html#preprocess-data-1"><i class="fa fa-check"></i><b>5.2</b> Preprocess data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Figure_5.html"><a href="05-Figure_5.html#colour-blindy-friendly-palettes"><i class="fa fa-check"></i><b>5.3</b> Colour blindy friendly palettes</a></li>
<li class="chapter" data-level="5.4" data-path="05-Figure_5.html"><a href="05-Figure_5.html#a-heatmap"><i class="fa fa-check"></i><b>5.4</b> A Heatmap</a></li>
<li class="chapter" data-level="5.5" data-path="05-Figure_5.html"><a href="05-Figure_5.html#b-c"><i class="fa fa-check"></i><b>5.5</b> B &amp; C</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="05-Figure_5.html"><a href="05-Figure_5.html#b-pcoa-plot"><i class="fa fa-check"></i><b>5.5.1</b> B PCoA plot</a></li>
<li class="chapter" data-level="5.5.2" data-path="05-Figure_5.html"><a href="05-Figure_5.html#c-pcoa-plot"><i class="fa fa-check"></i><b>5.5.2</b> C PCoA plot</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="05-Figure_5.html"><a href="05-Figure_5.html#without-ethanol"><i class="fa fa-check"></i><b>5.6</b> Without ethanol</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Figure_6.html"><a href="06-Figure_6.html"><i class="fa fa-check"></i><b>6</b> Figure 6</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Figure_6.html"><a href="06-Figure_6.html#libraries-3"><i class="fa fa-check"></i><b>6.1</b> Libraries</a></li>
<li class="chapter" data-level="6.2" data-path="06-Figure_6.html"><a href="06-Figure_6.html#preprocess-tables"><i class="fa fa-check"></i><b>6.2</b> Preprocess tables</a></li>
<li class="chapter" data-level="6.3" data-path="06-Figure_6.html"><a href="06-Figure_6.html#all-buffers"><i class="fa fa-check"></i><b>6.3</b> All buffers</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-Figure_6.html"><a href="06-Figure_6.html#correlation"><i class="fa fa-check"></i><b>6.3.1</b> Correlation</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-Figure_6.html"><a href="06-Figure_6.html#heatmap"><i class="fa fa-check"></i><b>6.3.2</b> Heatmap</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Supplementary figures</b></span></li>
<li class="chapter" data-level="7" data-path="07-Supp_fig_1.html"><a href="07-Supp_fig_1.html"><i class="fa fa-check"></i><b>7</b> Supp figure 1</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Supp_fig_1.html"><a href="07-Supp_fig_1.html#libraries-4"><i class="fa fa-check"></i><b>7.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2" data-path="07-Supp_fig_1.html"><a href="07-Supp_fig_1.html#rarefactipon-plot"><i class="fa fa-check"></i><b>7.2</b> Rarefactipon plot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html"><i class="fa fa-check"></i><b>8</b> Supp figure 2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#libraries-5"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#weighted-unifrac-plots-with-relative-abundance"><i class="fa fa-check"></i><b>8.2</b> Weighted unifrac plots with relative abundance</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-a-buffer-used-faceted-by-patient"><i class="fa fa-check"></i><b>8.2.1</b> Figure A: Buffer used faceted by patient</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-b-storage-temperature"><i class="fa fa-check"></i><b>8.2.2</b> Figure B: Storage temperature</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-c-storage-conditions"><i class="fa fa-check"></i><b>8.2.3</b> Figure C: Storage conditions</a></li>
<li class="chapter" data-level="8.2.4" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-d-buffer-and-storage-condition-faceted-by-patient"><i class="fa fa-check"></i><b>8.2.4</b> Figure D: Buffer and storage condition faceted by patient</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html"><i class="fa fa-check"></i><b>9</b> Supp figure 3</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#libraries-6"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#s-composition"><i class="fa fa-check"></i><b>9.2</b> 16S composition</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#table-of-all-paired-distances"><i class="fa fa-check"></i><b>9.2.1</b> Table of all paired distances</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#extract-the-paired-distances"><i class="fa fa-check"></i><b>9.2.2</b> Extract the paired distances</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#buffer-deep-dive"><i class="fa fa-check"></i><b>9.2.3</b> Buffer deep dive</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#combined-plot"><i class="fa fa-check"></i><b>9.2.4</b> Combined plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html"><i class="fa fa-check"></i><b>10</b> Supp figure 4</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#libraries-7"><i class="fa fa-check"></i><b>10.1</b> Libraries</a></li>
<li class="chapter" data-level="10.2" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#alpha-diveristy-plots"><i class="fa fa-check"></i><b>10.2</b> Alpha diveristy plots</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#a-storage-conditions"><i class="fa fa-check"></i><b>10.2.1</b> A: Storage conditions</a></li>
<li class="chapter" data-level="10.2.2" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#b-participant-number"><i class="fa fa-check"></i><b>10.2.2</b> B: Participant number</a></li>
<li class="chapter" data-level="10.2.3" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#c-storage-temp"><i class="fa fa-check"></i><b>10.2.3</b> C: Storage temp</a></li>
<li class="chapter" data-level="10.2.4" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#d-buffer-and-storage"><i class="fa fa-check"></i><b>10.2.4</b> D: Buffer and storage</a></li>
<li class="chapter" data-level="10.2.5" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#e-buffer-and-participant"><i class="fa fa-check"></i><b>10.2.5</b> E: Buffer and participant</a></li>
<li class="chapter" data-level="10.2.6" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#f-storage-condition-and-participant"><i class="fa fa-check"></i><b>10.2.6</b> F: storage condition and participant</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Supp_fig_5.html"><a href="11-Supp_fig_5.html"><i class="fa fa-check"></i><b>11</b> Supp figure 5</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Supp_fig_5.html"><a href="11-Supp_fig_5.html#libraries-8"><i class="fa fa-check"></i><b>11.1</b> Libraries</a></li>
<li class="chapter" data-level="11.2" data-path="11-Supp_fig_5.html"><a href="11-Supp_fig_5.html#barchart-of-genera-biomarkers"><i class="fa fa-check"></i><b>11.2</b> Barchart of Genera biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Supp_fig_6.html"><a href="12-Supp_fig_6.html"><i class="fa fa-check"></i><b>12</b> Supp figure 6</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Supp_fig_6.html"><a href="12-Supp_fig_6.html#intra-group-variability"><i class="fa fa-check"></i><b>12.1</b> Intra group variability</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html"><i class="fa fa-check"></i><b>13</b> Supp figure 7</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#libraries-9"><i class="fa fa-check"></i><b>13.1</b> Libraries</a></li>
<li class="chapter" data-level="13.2" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#scfa-analysis"><i class="fa fa-check"></i><b>13.2</b> SCFA analysis</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#table-of-all-paired-distances-1"><i class="fa fa-check"></i><b>13.2.1</b> Table of all paired distances</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#extract-the-paired-distances-1"><i class="fa fa-check"></i><b>13.2.2</b> Extract the paired distances</a></li>
<li class="chapter" data-level="13.2.3" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#buffer-deep-dive-1"><i class="fa fa-check"></i><b>13.2.3</b> Buffer deep dive</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#combine-plots"><i class="fa fa-check"></i><b>13.3</b> Combine plots</a></li>
</ul></li>
<li class="part"><span><b>Supplementary tables</b></span></li>
<li class="chapter" data-level="14" data-path="14-Supp_table_2.html"><a href="14-Supp_table_2.html"><i class="fa fa-check"></i><b>14</b> Supp table 2</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Supp_table_2.html"><a href="14-Supp_table_2.html#libraries-10"><i class="fa fa-check"></i><b>14.1</b> Libraries</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="14-Supp_table_2.html"><a href="14-Supp_table_2.html#statistical-test"><i class="fa fa-check"></i><b>14.1.1</b> Statistical test</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Supp_table_3.html"><a href="15-Supp_table_3.html"><i class="fa fa-check"></i><b>15</b> Supp table 3</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Supp_table_3.html"><a href="15-Supp_table_3.html#libraries-11"><i class="fa fa-check"></i><b>15.1</b> Libraries</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="15-Supp_table_3.html"><a href="15-Supp_table_3.html#paired-wilcox-tests"><i class="fa fa-check"></i><b>15.1.1</b> Paired wilcox tests</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Supp_table_4.html"><a href="16-Supp_table_4.html"><i class="fa fa-check"></i><b>16</b> Supp table 4</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Supp_table_4.html"><a href="16-Supp_table_4.html#libraries-12"><i class="fa fa-check"></i><b>16.1</b> Libraries</a></li>
<li class="chapter" data-level="16.2" data-path="16-Supp_table_4.html"><a href="16-Supp_table_4.html#significant-results-table"><i class="fa fa-check"></i><b>16.2</b> Significant results table</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Supp_table_6.html"><a href="17-Supp_table_6.html"><i class="fa fa-check"></i><b>17</b> Supp table 6</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Supp_table_6.html"><a href="17-Supp_table_6.html#libraries-13"><i class="fa fa-check"></i><b>17.1</b> libraries</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="17-Supp_table_6.html"><a href="17-Supp_table_6.html#statisitical-tests"><i class="fa fa-check"></i><b>17.1.1</b> Statisitical tests</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="18-Supp_table_7.html"><a href="18-Supp_table_7.html"><i class="fa fa-check"></i><b>18</b> Supp table 7</a></li>
<li class="chapter" data-level="19" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html"><i class="fa fa-check"></i><b>19</b> Supp table 8</a>
<ul>
<li class="chapter" data-level="19.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#libraries-14"><i class="fa fa-check"></i><b>19.1</b> Libraries</a></li>
<li class="chapter" data-level="19.2" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#preprocess-tables-1"><i class="fa fa-check"></i><b>19.2</b> Preprocess tables</a></li>
<li class="chapter" data-level="19.3" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#no-buffer"><i class="fa fa-check"></i><b>19.3</b> No buffer</a>
<ul>
<li class="chapter" data-level="19.3.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-1"><i class="fa fa-check"></i><b>19.3.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.4" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#rnalater"><i class="fa fa-check"></i><b>19.4</b> RNAlater</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-2"><i class="fa fa-check"></i><b>19.4.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#ethanol"><i class="fa fa-check"></i><b>19.5</b> Ethanol</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-3"><i class="fa fa-check"></i><b>19.5.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#psp"><i class="fa fa-check"></i><b>19.6</b> PSP</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-4"><i class="fa fa-check"></i><b>19.6.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#compare-results"><i class="fa fa-check"></i><b>19.7</b> Compare results</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Optimised human stool sample collection for multi-omic microbiota analysis R code</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-and-preocessing-data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Importing and preocessing data<a href="02-Phyloseqize.html#importing-and-preocessing-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="qiime2r-import" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> qiime2R import<a href="02-Phyloseqize.html#qiime2r-import" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="02-Phyloseqize.html#cb1-1" tabindex="-1"></a><span class="co">#Load qiime2R library</span></span>
<span id="cb1-2"><a href="02-Phyloseqize.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(qiime2R)</span>
<span id="cb1-3"><a href="02-Phyloseqize.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-4"><a href="02-Phyloseqize.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-5"><a href="02-Phyloseqize.html#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="02-Phyloseqize.html#cb1-6" tabindex="-1"></a><span class="co">#Convert QIIME2 artifacts into phyloseq object</span></span>
<span id="cb1-7"><a href="02-Phyloseqize.html#cb1-7" tabindex="-1"></a>physeq<span class="ot">&lt;-</span><span class="fu">qza_to_phyloseq</span>(</span>
<span id="cb1-8"><a href="02-Phyloseqize.html#cb1-8" tabindex="-1"></a>  <span class="at">features=</span><span class="st">&quot;./data/table.nocontrol.minfreq4628.qza&quot;</span>,</span>
<span id="cb1-9"><a href="02-Phyloseqize.html#cb1-9" tabindex="-1"></a>  <span class="at">tree=</span><span class="st">&quot;./data/rooted-tree.qza&quot;</span>,</span>
<span id="cb1-10"><a href="02-Phyloseqize.html#cb1-10" tabindex="-1"></a>  <span class="at">taxonomy=</span><span class="st">&quot;./data/taxonomy.qza&quot;</span>,</span>
<span id="cb1-11"><a href="02-Phyloseqize.html#cb1-11" tabindex="-1"></a>  <span class="at">metadata =</span> <span class="st">&quot;./data/qiime2_metadata.nocontrols.v4.tsv&quot;</span></span>
<span id="cb1-12"><a href="02-Phyloseqize.html#cb1-12" tabindex="-1"></a>)</span>
<span id="cb1-13"><a href="02-Phyloseqize.html#cb1-13" tabindex="-1"></a></span>
<span id="cb1-14"><a href="02-Phyloseqize.html#cb1-14" tabindex="-1"></a><span class="co">#Some oRikenellaceae are given the genera human and the species gut metagenome</span></span>
<span id="cb1-15"><a href="02-Phyloseqize.html#cb1-15" tabindex="-1"></a><span class="co">#Change genera &quot;human&quot; to NA</span></span>
<span id="cb1-16"><a href="02-Phyloseqize.html#cb1-16" tabindex="-1"></a><span class="co">#change species &quot;gut metagenome&quot; to NA</span></span>
<span id="cb1-17"><a href="02-Phyloseqize.html#cb1-17" tabindex="-1"></a><span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>], <span class="at">pattern =</span> <span class="st">&quot;human&quot;</span>, <span class="at">replacement =</span> <span class="cn">NA</span>)</span>
<span id="cb1-18"><a href="02-Phyloseqize.html#cb1-18" tabindex="-1"></a><span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Species&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Species&quot;</span>], <span class="at">pattern =</span> <span class="st">&quot;gut metagenome&quot;</span>, <span class="at">replacement =</span> <span class="cn">NA</span>)</span>
<span id="cb1-19"><a href="02-Phyloseqize.html#cb1-19" tabindex="-1"></a></span>
<span id="cb1-20"><a href="02-Phyloseqize.html#cb1-20" tabindex="-1"></a><span class="co">#save physeq object</span></span>
<span id="cb1-21"><a href="02-Phyloseqize.html#cb1-21" tabindex="-1"></a><span class="fu">save</span>(physeq, <span class="at">file =</span> <span class="st">&quot;./data/physeq&quot;</span>)</span></code></pre></div>
</div>
<div id="preprocess-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Preprocess data<a href="02-Phyloseqize.html#preprocess-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="02-Phyloseqize.html#cb2-1" tabindex="-1"></a><span class="co">#libraries</span></span>
<span id="cb2-2"><a href="02-Phyloseqize.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb2-3"><a href="02-Phyloseqize.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;microbiome&quot;</span>)</span>
<span id="cb2-4"><a href="02-Phyloseqize.html#cb2-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb2-5"><a href="02-Phyloseqize.html#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="02-Phyloseqize.html#cb2-6" tabindex="-1"></a><span class="co">#load in phyloseq object</span></span>
<span id="cb2-7"><a href="02-Phyloseqize.html#cb2-7" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;./data/physeq&quot;</span>)</span>
<span id="cb2-8"><a href="02-Phyloseqize.html#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="02-Phyloseqize.html#cb2-9" tabindex="-1"></a><span class="co">#subset samples to only keep rnalater samples</span></span>
<span id="cb2-10"><a href="02-Phyloseqize.html#cb2-10" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(physeq, <span class="sc">!</span>(RNAlater_washed_status <span class="sc">==</span> <span class="st">&quot;unwashed&quot;</span>))</span>
<span id="cb2-11"><a href="02-Phyloseqize.html#cb2-11" tabindex="-1"></a></span>
<span id="cb2-12"><a href="02-Phyloseqize.html#cb2-12" tabindex="-1"></a><span class="co">#Change names of RNA later samples</span></span>
<span id="cb2-13"><a href="02-Phyloseqize.html#cb2-13" tabindex="-1"></a>sample_name_vec <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">as.vector</span>(<span class="fu">sample_data</span>(physeq)[,<span class="dv">1</span>]))</span>
<span id="cb2-14"><a href="02-Phyloseqize.html#cb2-14" tabindex="-1"></a>sample_name_vec_corrected <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;_b&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>,sample_name_vec)</span>
<span id="cb2-15"><a href="02-Phyloseqize.html#cb2-15" tabindex="-1"></a><span class="fu">sample_data</span>(physeq)[,<span class="dv">1</span>] <span class="ot">&lt;-</span> sample_name_vec_corrected</span>
<span id="cb2-16"><a href="02-Phyloseqize.html#cb2-16" tabindex="-1"></a></span>
<span id="cb2-17"><a href="02-Phyloseqize.html#cb2-17" tabindex="-1"></a><span class="co">#In metadata add column with buffer and storage combined</span></span>
<span id="cb2-18"><a href="02-Phyloseqize.html#cb2-18" tabindex="-1"></a><span class="fu">sample_data</span>(physeq)[,<span class="st">&quot;Bufferused_and_Storageconditions&quot;</span>] <span class="ot">&lt;-</span> </span>
<span id="cb2-19"><a href="02-Phyloseqize.html#cb2-19" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="fu">unlist</span>(<span class="fu">sample_data</span>(physeq)[,<span class="st">&quot;Bufferused&quot;</span>]),</span>
<span id="cb2-20"><a href="02-Phyloseqize.html#cb2-20" tabindex="-1"></a>         <span class="st">&quot;_&quot;</span>,</span>
<span id="cb2-21"><a href="02-Phyloseqize.html#cb2-21" tabindex="-1"></a>         <span class="fu">unlist</span>(<span class="fu">sample_data</span>(physeq)[,<span class="st">&quot;Storageconditions&quot;</span>]))</span>
<span id="cb2-22"><a href="02-Phyloseqize.html#cb2-22" tabindex="-1"></a></span>
<span id="cb2-23"><a href="02-Phyloseqize.html#cb2-23" tabindex="-1"></a><span class="fu">save</span>(physeq, <span class="at">file =</span> <span class="st">&quot;./data/preprocess_physeq&quot;</span>)</span>
<span id="cb2-24"><a href="02-Phyloseqize.html#cb2-24" tabindex="-1"></a></span>
<span id="cb2-25"><a href="02-Phyloseqize.html#cb2-25" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb2-26"><a href="02-Phyloseqize.html#cb2-26" tabindex="-1"></a></span>
<span id="cb2-27"><a href="02-Phyloseqize.html#cb2-27" tabindex="-1"></a><span class="co"># preprocess ####</span></span>
<span id="cb2-28"><a href="02-Phyloseqize.html#cb2-28" tabindex="-1"></a><span class="co">#transform samples to relabund</span></span>
<span id="cb2-29"><a href="02-Phyloseqize.html#cb2-29" tabindex="-1"></a>physeq_relabund <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(physeq, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb2-30"><a href="02-Phyloseqize.html#cb2-30" tabindex="-1"></a><span class="co">#num ASVs b4 removal of low ASVs = 3965</span></span>
<span id="cb2-31"><a href="02-Phyloseqize.html#cb2-31" tabindex="-1"></a><span class="co"># nrow(otu_table(physeq_relabund))</span></span>
<span id="cb2-32"><a href="02-Phyloseqize.html#cb2-32" tabindex="-1"></a><span class="co">#first remove ASV with relabund equal to 0</span></span>
<span id="cb2-33"><a href="02-Phyloseqize.html#cb2-33" tabindex="-1"></a>physeq_relabund <span class="ot">&lt;-</span>  <span class="fu">filter_taxa</span>(physeq_relabund, <span class="cf">function</span>(x) <span class="fu">sum</span>(x) <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="cn">TRUE</span>)</span>
<span id="cb2-34"><a href="02-Phyloseqize.html#cb2-34" tabindex="-1"></a><span class="co">#current num ASVs = 3626</span></span>
<span id="cb2-35"><a href="02-Phyloseqize.html#cb2-35" tabindex="-1"></a><span class="co"># nrow(otu_table(physeq_relabund))</span></span>
<span id="cb2-36"><a href="02-Phyloseqize.html#cb2-36" tabindex="-1"></a><span class="co">#remove ASVs with a mean less than 1e-5 (relabund)</span></span>
<span id="cb2-37"><a href="02-Phyloseqize.html#cb2-37" tabindex="-1"></a>physeq_relabund <span class="ot">&lt;-</span>  <span class="fu">filter_taxa</span>(physeq_relabund, <span class="cf">function</span>(x) <span class="fu">mean</span>(x) <span class="sc">&gt;</span> <span class="fl">1e-5</span>, <span class="cn">TRUE</span>)</span>
<span id="cb2-38"><a href="02-Phyloseqize.html#cb2-38" tabindex="-1"></a><span class="co">#current num ASVs = 1406</span></span>
<span id="cb2-39"><a href="02-Phyloseqize.html#cb2-39" tabindex="-1"></a><span class="co"># nrow(otu_table(physeq_relabund))</span></span>
<span id="cb2-40"><a href="02-Phyloseqize.html#cb2-40" tabindex="-1"></a><span class="co">#Let us see how much relabund we lost</span></span>
<span id="cb2-41"><a href="02-Phyloseqize.html#cb2-41" tabindex="-1"></a><span class="co">#colSums(otu_table(physeq_relabund))</span></span>
<span id="cb2-42"><a href="02-Phyloseqize.html#cb2-42" tabindex="-1"></a><span class="co">#all samples have greater than 0.98 relabund remaining so it looks good</span></span>
<span id="cb2-43"><a href="02-Phyloseqize.html#cb2-43" tabindex="-1"></a></span>
<span id="cb2-44"><a href="02-Phyloseqize.html#cb2-44" tabindex="-1"></a><span class="co">#In metadata add column with buffer and storage combined</span></span>
<span id="cb2-45"><a href="02-Phyloseqize.html#cb2-45" tabindex="-1"></a><span class="fu">sample_data</span>(physeq_relabund)[,<span class="st">&quot;Bufferused_and_Storageconditions&quot;</span>] <span class="ot">&lt;-</span> </span>
<span id="cb2-46"><a href="02-Phyloseqize.html#cb2-46" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="fu">unlist</span>(<span class="fu">sample_data</span>(physeq_relabund)[,<span class="st">&quot;Bufferused&quot;</span>]),</span>
<span id="cb2-47"><a href="02-Phyloseqize.html#cb2-47" tabindex="-1"></a>         <span class="st">&quot;_&quot;</span>,</span>
<span id="cb2-48"><a href="02-Phyloseqize.html#cb2-48" tabindex="-1"></a>         <span class="fu">unlist</span>(<span class="fu">sample_data</span>(physeq_relabund)[,<span class="st">&quot;Storageconditions&quot;</span>]))</span>
<span id="cb2-49"><a href="02-Phyloseqize.html#cb2-49" tabindex="-1"></a><span class="co">#save physeq object</span></span>
<span id="cb2-50"><a href="02-Phyloseqize.html#cb2-50" tabindex="-1"></a><span class="fu">save</span>(physeq_relabund, <span class="at">file =</span> <span class="st">&quot;./data/physeq_relabund&quot;</span>)</span></code></pre></div>
</div>
<div id="lipid-data" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Lipid data<a href="02-Phyloseqize.html#lipid-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="02-Phyloseqize.html#cb3-1" tabindex="-1"></a><span class="co">#libraries</span></span>
<span id="cb3-2"><a href="02-Phyloseqize.html#cb3-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phyloseq&quot;</span>)</span>
<span id="cb3-3"><a href="02-Phyloseqize.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="02-Phyloseqize.html#cb3-4" tabindex="-1"></a><span class="co">#Read in data</span></span>
<span id="cb3-5"><a href="02-Phyloseqize.html#cb3-5" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Pilot SCFA data - MG.csv&quot;</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-6"><a href="02-Phyloseqize.html#cb3-6" tabindex="-1"></a>                <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb3-7"><a href="02-Phyloseqize.html#cb3-7" tabindex="-1"></a><span class="co">#Sample names</span></span>
<span id="cb3-8"><a href="02-Phyloseqize.html#cb3-8" tabindex="-1"></a>sample_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)</span>
<span id="cb3-9"><a href="02-Phyloseqize.html#cb3-9" tabindex="-1"></a><span class="co">#compounds</span></span>
<span id="cb3-10"><a href="02-Phyloseqize.html#cb3-10" tabindex="-1"></a>compound_names <span class="ot">&lt;-</span> <span class="fu">row.names</span>(df[<span class="dv">6</span><span class="sc">:</span><span class="dv">14</span>,])</span>
<span id="cb3-11"><a href="02-Phyloseqize.html#cb3-11" tabindex="-1"></a></span>
<span id="cb3-12"><a href="02-Phyloseqize.html#cb3-12" tabindex="-1"></a></span>
<span id="cb3-13"><a href="02-Phyloseqize.html#cb3-13" tabindex="-1"></a><span class="co">#First create OTU matrix</span></span>
<span id="cb3-14"><a href="02-Phyloseqize.html#cb3-14" tabindex="-1"></a>otu_df <span class="ot">&lt;-</span> df[<span class="dv">6</span><span class="sc">:</span><span class="dv">14</span>,]</span>
<span id="cb3-15"><a href="02-Phyloseqize.html#cb3-15" tabindex="-1"></a>num_otu_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(otu_df, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x))))</span>
<span id="cb3-16"><a href="02-Phyloseqize.html#cb3-16" tabindex="-1"></a>otu_mat <span class="ot">&lt;-</span> <span class="fu">data.matrix</span>(num_otu_df)</span>
<span id="cb3-17"><a href="02-Phyloseqize.html#cb3-17" tabindex="-1"></a><span class="fu">rownames</span>(otu_mat) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;OTU&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(otu_mat))</span>
<span id="cb3-18"><a href="02-Phyloseqize.html#cb3-18" tabindex="-1"></a><span class="fu">colnames</span>(otu_mat) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Sample&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(otu_mat))</span>
<span id="cb3-19"><a href="02-Phyloseqize.html#cb3-19" tabindex="-1"></a></span>
<span id="cb3-20"><a href="02-Phyloseqize.html#cb3-20" tabindex="-1"></a><span class="co">#Tax matrix</span></span>
<span id="cb3-21"><a href="02-Phyloseqize.html#cb3-21" tabindex="-1"></a>tax_mat <span class="ot">&lt;-</span> otu_mat[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb3-22"><a href="02-Phyloseqize.html#cb3-22" tabindex="-1"></a>tax_mat[,<span class="dv">1</span>] <span class="ot">&lt;-</span> compound_names</span>
<span id="cb3-23"><a href="02-Phyloseqize.html#cb3-23" tabindex="-1"></a>tax_mat[,<span class="dv">2</span>] <span class="ot">&lt;-</span> compound_names</span>
<span id="cb3-24"><a href="02-Phyloseqize.html#cb3-24" tabindex="-1"></a><span class="fu">colnames</span>(tax_mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Compound&quot;</span>,<span class="st">&quot;X&quot;</span>)</span>
<span id="cb3-25"><a href="02-Phyloseqize.html#cb3-25" tabindex="-1"></a></span>
<span id="cb3-26"><a href="02-Phyloseqize.html#cb3-26" tabindex="-1"></a><span class="co">#Sample_data_frame</span></span>
<span id="cb3-27"><a href="02-Phyloseqize.html#cb3-27" tabindex="-1"></a>metadf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">lapply</span>(df[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,], as.character), <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb3-28"><a href="02-Phyloseqize.html#cb3-28" tabindex="-1"></a><span class="fu">row.names</span>(metadf) <span class="ot">&lt;-</span> <span class="fu">row.names</span>(df[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,])</span>
<span id="cb3-29"><a href="02-Phyloseqize.html#cb3-29" tabindex="-1"></a>metadf[<span class="st">&quot;Sample_name&quot;</span>,] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)</span>
<span id="cb3-30"><a href="02-Phyloseqize.html#cb3-30" tabindex="-1"></a><span class="co">#Change buffer labels</span></span>
<span id="cb3-31"><a href="02-Phyloseqize.html#cb3-31" tabindex="-1"></a>metadf[<span class="dv">2</span>,] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;None&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;No buffer&quot;</span>, metadf[<span class="dv">2</span>,])</span>
<span id="cb3-32"><a href="02-Phyloseqize.html#cb3-32" tabindex="-1"></a>metadf[<span class="dv">2</span>,] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;70% EtOH&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;Ethanol&quot;</span>, metadf[<span class="dv">2</span>,])</span>
<span id="cb3-33"><a href="02-Phyloseqize.html#cb3-33" tabindex="-1"></a>metadf[<span class="dv">2</span>,] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;RNA later&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;RNAlater&quot;</span>, metadf[<span class="dv">2</span>,])</span>
<span id="cb3-34"><a href="02-Phyloseqize.html#cb3-34" tabindex="-1"></a></span>
<span id="cb3-35"><a href="02-Phyloseqize.html#cb3-35" tabindex="-1"></a><span class="fu">colnames</span>(metadf) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(otu_mat)</span>
<span id="cb3-36"><a href="02-Phyloseqize.html#cb3-36" tabindex="-1"></a>t_metadf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(metadf), <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb3-37"><a href="02-Phyloseqize.html#cb3-37" tabindex="-1"></a>t_metadf[t_metadf<span class="sc">$</span>Storage.time <span class="sc">==</span> <span class="dv">0</span>,<span class="st">&quot;Temp&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Baseline&quot;</span></span>
<span id="cb3-38"><a href="02-Phyloseqize.html#cb3-38" tabindex="-1"></a>t_metadf[,<span class="st">&quot;Buffer.type.Temp&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(t_metadf[,<span class="st">&quot;Buffer.type&quot;</span>],t_metadf[,<span class="st">&quot;Temp&quot;</span>],<span class="at">sep=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb3-39"><a href="02-Phyloseqize.html#cb3-39" tabindex="-1"></a><span class="co">#Temp and time column</span></span>
<span id="cb3-40"><a href="02-Phyloseqize.html#cb3-40" tabindex="-1"></a>t_metadf[,<span class="st">&quot;Temp.Storage.time&quot;</span>] <span class="ot">&lt;-</span> </span>
<span id="cb3-41"><a href="02-Phyloseqize.html#cb3-41" tabindex="-1"></a>  <span class="fu">paste</span>(t_metadf[,<span class="st">&quot;Temp&quot;</span>],t_metadf[,<span class="st">&quot;Storage.time&quot;</span>],<span class="at">sep=</span><span class="st">&quot;|&quot;</span>)</span>
<span id="cb3-42"><a href="02-Phyloseqize.html#cb3-42" tabindex="-1"></a><span class="co">#Change Baseline|0 to Baseline</span></span>
<span id="cb3-43"><a href="02-Phyloseqize.html#cb3-43" tabindex="-1"></a>t_metadf[,<span class="st">&quot;Temp.Storage.time&quot;</span>] <span class="ot">&lt;-</span> </span>
<span id="cb3-44"><a href="02-Phyloseqize.html#cb3-44" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;Baseline</span><span class="sc">\\</span><span class="st">|0&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;Baseline&quot;</span>, </span>
<span id="cb3-45"><a href="02-Phyloseqize.html#cb3-45" tabindex="-1"></a>       <span class="at">x =</span> t_metadf[,<span class="st">&quot;Temp.Storage.time&quot;</span>])</span>
<span id="cb3-46"><a href="02-Phyloseqize.html#cb3-46" tabindex="-1"></a><span class="co">#Order of temp</span></span>
<span id="cb3-47"><a href="02-Phyloseqize.html#cb3-47" tabindex="-1"></a>t_metadf[,<span class="st">&quot;Temp&quot;</span>] <span class="ot">&lt;-</span> </span>
<span id="cb3-48"><a href="02-Phyloseqize.html#cb3-48" tabindex="-1"></a>  <span class="fu">factor</span>(t_metadf[,<span class="st">&quot;Temp&quot;</span>], <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Baseline&quot;</span>, <span class="st">&quot;RT&quot;</span>, <span class="st">&quot;Fridge&quot;</span>))</span>
<span id="cb3-49"><a href="02-Phyloseqize.html#cb3-49" tabindex="-1"></a><span class="co">#Sample column</span></span>
<span id="cb3-50"><a href="02-Phyloseqize.html#cb3-50" tabindex="-1"></a>t_metadf<span class="sc">$</span><span class="st">`</span><span class="at">Sample name</span><span class="st">`</span> <span class="ot">&lt;-</span> t_metadf<span class="sc">$</span>Sample_name</span>
<span id="cb3-51"><a href="02-Phyloseqize.html#cb3-51" tabindex="-1"></a></span>
<span id="cb3-52"><a href="02-Phyloseqize.html#cb3-52" tabindex="-1"></a></span>
<span id="cb3-53"><a href="02-Phyloseqize.html#cb3-53" tabindex="-1"></a><span class="co">#Create phyloseq object</span></span>
<span id="cb3-54"><a href="02-Phyloseqize.html#cb3-54" tabindex="-1"></a>OTU <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(otu_mat, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-55"><a href="02-Phyloseqize.html#cb3-55" tabindex="-1"></a>TAX <span class="ot">&lt;-</span> <span class="fu">tax_table</span>(tax_mat)</span>
<span id="cb3-56"><a href="02-Phyloseqize.html#cb3-56" tabindex="-1"></a>sampledata <span class="ot">&lt;-</span> <span class="fu">sample_data</span>(<span class="fu">data.frame</span>(t_metadf, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>))</span>
<span id="cb3-57"><a href="02-Phyloseqize.html#cb3-57" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(OTU,TAX,sampledata)</span>
<span id="cb3-58"><a href="02-Phyloseqize.html#cb3-58" tabindex="-1"></a></span>
<span id="cb3-59"><a href="02-Phyloseqize.html#cb3-59" tabindex="-1"></a><span class="co">#remove samples with no otu numbers</span></span>
<span id="cb3-60"><a href="02-Phyloseqize.html#cb3-60" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">subset_samples</span>(physeq, Sample_name <span class="sc">!=</span> <span class="st">&quot;6Q&quot;</span>)</span>
<span id="cb3-61"><a href="02-Phyloseqize.html#cb3-61" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">subset_samples</span>(physeq, Sample_name <span class="sc">!=</span> <span class="st">&quot;6R&quot;</span>)</span>
<span id="cb3-62"><a href="02-Phyloseqize.html#cb3-62" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">subset_samples</span>(physeq, Sample_name <span class="sc">!=</span> <span class="st">&quot;6S&quot;</span>)</span>
<span id="cb3-63"><a href="02-Phyloseqize.html#cb3-63" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">subset_samples</span>(physeq, Sample_name <span class="sc">!=</span> <span class="st">&quot;6T&quot;</span>)</span>
<span id="cb3-64"><a href="02-Phyloseqize.html#cb3-64" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">subset_samples</span>(physeq, Sample_name <span class="sc">!=</span> <span class="st">&quot;6P&quot;</span>)</span>
<span id="cb3-65"><a href="02-Phyloseqize.html#cb3-65" tabindex="-1"></a></span>
<span id="cb3-66"><a href="02-Phyloseqize.html#cb3-66" tabindex="-1"></a><span class="co">#save physeq object</span></span>
<span id="cb3-67"><a href="02-Phyloseqize.html#cb3-67" tabindex="-1"></a><span class="fu">save</span>(physeq, <span class="at">file =</span> <span class="st">&quot;data/lipid_physeq_object&quot;</span>)</span></code></pre></div>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="01-Intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="03-Figure_3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
