<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Figure 6 | Optimised human stool sample collection for multi-omic microbiota analysis R code</title>
  <meta name="description" content="Book containg the R code for the paper Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Figure 6 | Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Book containg the R code for the paper Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Figure 6 | Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  
  <meta name="twitter:description" content="Book containg the R code for the paper Optimised human stool sample collection for multi-omic microbiota analysis R code" />
  

<meta name="author" content="Matthew R. Gemmell" />


<meta name="date" content="2024-07-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="05-Figure_5.html"/>
<link rel="next" href="07-Supp_fig_1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a>Optimised human stool sample collection for multi-omic microbiota analysis R code</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="01-Intro.html"><a href="01-Intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html"><i class="fa fa-check"></i><b>2</b> Importing and preocessing data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html#qiime2r-import"><i class="fa fa-check"></i><b>2.1</b> qiime2R import</a></li>
<li class="chapter" data-level="2.2" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html#preprocess-data"><i class="fa fa-check"></i><b>2.2</b> Preprocess data</a></li>
<li class="chapter" data-level="2.3" data-path="02-Phyloseqize.html"><a href="02-Phyloseqize.html#lipid-data"><i class="fa fa-check"></i><b>2.3</b> Lipid data</a></li>
</ul></li>
<li class="part"><span><b>Main figures</b></span></li>
<li class="chapter" data-level="3" data-path="03-Figure_3.html"><a href="03-Figure_3.html"><i class="fa fa-check"></i><b>3</b> Figure 3</a>
<ul>
<li class="chapter" data-level="3.1" data-path="03-Figure_3.html"><a href="03-Figure_3.html#libraries"><i class="fa fa-check"></i><b>3.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2" data-path="03-Figure_3.html"><a href="03-Figure_3.html#figure-3a"><i class="fa fa-check"></i><b>3.2</b> Figure 3A</a></li>
<li class="chapter" data-level="3.3" data-path="03-Figure_3.html"><a href="03-Figure_3.html#figure-3b"><i class="fa fa-check"></i><b>3.3</b> Figure 3B</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="03-Figure_3.html"><a href="03-Figure_3.html#percentage-of-bacteroides-and-firmicutes"><i class="fa fa-check"></i><b>3.3.1</b> Percentage of Bacteroides and Firmicutes</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="03-Figure_3.html"><a href="03-Figure_3.html#figure-3c"><i class="fa fa-check"></i><b>3.4</b> Figure 3C</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="04-Figure_4.html"><a href="04-Figure_4.html"><i class="fa fa-check"></i><b>4</b> Figure 4</a>
<ul>
<li class="chapter" data-level="4.1" data-path="04-Figure_4.html"><a href="04-Figure_4.html#libraries-1"><i class="fa fa-check"></i><b>4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2" data-path="04-Figure_4.html"><a href="04-Figure_4.html#genus"><i class="fa fa-check"></i><b>4.2</b> Genus</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="04-Figure_4.html"><a href="04-Figure_4.html#preprocess-table-and-convert-to-genus-table"><i class="fa fa-check"></i><b>4.2.1</b> Preprocess table and convert to genus table</a></li>
<li class="chapter" data-level="4.2.2" data-path="04-Figure_4.html"><a href="04-Figure_4.html#maaslin2-on-genera"><i class="fa fa-check"></i><b>4.2.2</b> MaAsLin2 on genera</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="05-Figure_5.html"><a href="05-Figure_5.html"><i class="fa fa-check"></i><b>5</b> Figure 5</a>
<ul>
<li class="chapter" data-level="5.1" data-path="05-Figure_5.html"><a href="05-Figure_5.html#libraries-2"><i class="fa fa-check"></i><b>5.1</b> libraries</a></li>
<li class="chapter" data-level="5.2" data-path="05-Figure_5.html"><a href="05-Figure_5.html#preprocess-data-1"><i class="fa fa-check"></i><b>5.2</b> Preprocess data</a></li>
<li class="chapter" data-level="5.3" data-path="05-Figure_5.html"><a href="05-Figure_5.html#colour-blindy-friendly-palettes"><i class="fa fa-check"></i><b>5.3</b> Colour blindy friendly palettes</a></li>
<li class="chapter" data-level="5.4" data-path="05-Figure_5.html"><a href="05-Figure_5.html#a-heatmap"><i class="fa fa-check"></i><b>5.4</b> A Heatmap</a></li>
<li class="chapter" data-level="5.5" data-path="05-Figure_5.html"><a href="05-Figure_5.html#b-c"><i class="fa fa-check"></i><b>5.5</b> B &amp; C</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="05-Figure_5.html"><a href="05-Figure_5.html#b-pcoa-plot"><i class="fa fa-check"></i><b>5.5.1</b> B PCoA plot</a></li>
<li class="chapter" data-level="5.5.2" data-path="05-Figure_5.html"><a href="05-Figure_5.html#c-pcoa-plot"><i class="fa fa-check"></i><b>5.5.2</b> C PCoA plot</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="05-Figure_5.html"><a href="05-Figure_5.html#without-ethanol"><i class="fa fa-check"></i><b>5.6</b> Without ethanol</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="06-Figure_6.html"><a href="06-Figure_6.html"><i class="fa fa-check"></i><b>6</b> Figure 6</a>
<ul>
<li class="chapter" data-level="6.1" data-path="06-Figure_6.html"><a href="06-Figure_6.html#libraries-3"><i class="fa fa-check"></i><b>6.1</b> Libraries</a></li>
<li class="chapter" data-level="6.2" data-path="06-Figure_6.html"><a href="06-Figure_6.html#preprocess-tables"><i class="fa fa-check"></i><b>6.2</b> Preprocess tables</a></li>
<li class="chapter" data-level="6.3" data-path="06-Figure_6.html"><a href="06-Figure_6.html#all-buffers"><i class="fa fa-check"></i><b>6.3</b> All buffers</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="06-Figure_6.html"><a href="06-Figure_6.html#correlation"><i class="fa fa-check"></i><b>6.3.1</b> Correlation</a></li>
<li class="chapter" data-level="6.3.2" data-path="06-Figure_6.html"><a href="06-Figure_6.html#heatmap"><i class="fa fa-check"></i><b>6.3.2</b> Heatmap</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Supplementary figures</b></span></li>
<li class="chapter" data-level="7" data-path="07-Supp_fig_1.html"><a href="07-Supp_fig_1.html"><i class="fa fa-check"></i><b>7</b> Supp figure 1</a>
<ul>
<li class="chapter" data-level="7.1" data-path="07-Supp_fig_1.html"><a href="07-Supp_fig_1.html#libraries-4"><i class="fa fa-check"></i><b>7.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2" data-path="07-Supp_fig_1.html"><a href="07-Supp_fig_1.html#rarefactipon-plot"><i class="fa fa-check"></i><b>7.2</b> Rarefactipon plot</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html"><i class="fa fa-check"></i><b>8</b> Supp figure 2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#libraries-5"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#weighted-unifrac-plots-with-relative-abundance"><i class="fa fa-check"></i><b>8.2</b> Weighted unifrac plots with relative abundance</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-a-buffer-used-faceted-by-patient"><i class="fa fa-check"></i><b>8.2.1</b> Figure A: Buffer used faceted by patient</a></li>
<li class="chapter" data-level="8.2.2" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-b-storage-temperature"><i class="fa fa-check"></i><b>8.2.2</b> Figure B: Storage temperature</a></li>
<li class="chapter" data-level="8.2.3" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-c-storage-conditions"><i class="fa fa-check"></i><b>8.2.3</b> Figure C: Storage conditions</a></li>
<li class="chapter" data-level="8.2.4" data-path="08-Supp_fig_2.html"><a href="08-Supp_fig_2.html#figure-d-buffer-and-storage-condition-faceted-by-patient"><i class="fa fa-check"></i><b>8.2.4</b> Figure D: Buffer and storage condition faceted by patient</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html"><i class="fa fa-check"></i><b>9</b> Supp figure 3</a>
<ul>
<li class="chapter" data-level="9.1" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#libraries-6"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#s-composition"><i class="fa fa-check"></i><b>9.2</b> 16S composition</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#table-of-all-paired-distances"><i class="fa fa-check"></i><b>9.2.1</b> Table of all paired distances</a></li>
<li class="chapter" data-level="9.2.2" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#extract-the-paired-distances"><i class="fa fa-check"></i><b>9.2.2</b> Extract the paired distances</a></li>
<li class="chapter" data-level="9.2.3" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#buffer-deep-dive"><i class="fa fa-check"></i><b>9.2.3</b> Buffer deep dive</a></li>
<li class="chapter" data-level="9.2.4" data-path="09-Supp_fig_3.html"><a href="09-Supp_fig_3.html#combined-plot"><i class="fa fa-check"></i><b>9.2.4</b> Combined plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html"><i class="fa fa-check"></i><b>10</b> Supp figure 4</a>
<ul>
<li class="chapter" data-level="10.1" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#libraries-7"><i class="fa fa-check"></i><b>10.1</b> Libraries</a></li>
<li class="chapter" data-level="10.2" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#alpha-diveristy-plots"><i class="fa fa-check"></i><b>10.2</b> Alpha diveristy plots</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#a-storage-conditions"><i class="fa fa-check"></i><b>10.2.1</b> A: Storage conditions</a></li>
<li class="chapter" data-level="10.2.2" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#b-participant-number"><i class="fa fa-check"></i><b>10.2.2</b> B: Participant number</a></li>
<li class="chapter" data-level="10.2.3" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#c-storage-temp"><i class="fa fa-check"></i><b>10.2.3</b> C: Storage temp</a></li>
<li class="chapter" data-level="10.2.4" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#d-buffer-and-storage"><i class="fa fa-check"></i><b>10.2.4</b> D: Buffer and storage</a></li>
<li class="chapter" data-level="10.2.5" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#e-buffer-and-participant"><i class="fa fa-check"></i><b>10.2.5</b> E: Buffer and participant</a></li>
<li class="chapter" data-level="10.2.6" data-path="10-Supp_fig_4.html"><a href="10-Supp_fig_4.html#f-storage-condition-and-participant"><i class="fa fa-check"></i><b>10.2.6</b> F: storage condition and participant</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="11-Supp_fig_5.html"><a href="11-Supp_fig_5.html"><i class="fa fa-check"></i><b>11</b> Supp figure 5</a>
<ul>
<li class="chapter" data-level="11.1" data-path="11-Supp_fig_5.html"><a href="11-Supp_fig_5.html#libraries-8"><i class="fa fa-check"></i><b>11.1</b> Libraries</a></li>
<li class="chapter" data-level="11.2" data-path="11-Supp_fig_5.html"><a href="11-Supp_fig_5.html#barchart-of-genera-biomarkers"><i class="fa fa-check"></i><b>11.2</b> Barchart of Genera biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="12-Supp_fig_6.html"><a href="12-Supp_fig_6.html"><i class="fa fa-check"></i><b>12</b> Supp figure 6</a>
<ul>
<li class="chapter" data-level="12.1" data-path="12-Supp_fig_6.html"><a href="12-Supp_fig_6.html#intra-group-variability"><i class="fa fa-check"></i><b>12.1</b> Intra group variability</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html"><i class="fa fa-check"></i><b>13</b> Supp figure 7</a>
<ul>
<li class="chapter" data-level="13.1" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#libraries-9"><i class="fa fa-check"></i><b>13.1</b> Libraries</a></li>
<li class="chapter" data-level="13.2" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#scfa-analysis"><i class="fa fa-check"></i><b>13.2</b> SCFA analysis</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#table-of-all-paired-distances-1"><i class="fa fa-check"></i><b>13.2.1</b> Table of all paired distances</a></li>
<li class="chapter" data-level="13.2.2" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#extract-the-paired-distances-1"><i class="fa fa-check"></i><b>13.2.2</b> Extract the paired distances</a></li>
<li class="chapter" data-level="13.2.3" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#buffer-deep-dive-1"><i class="fa fa-check"></i><b>13.2.3</b> Buffer deep dive</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="13-Supp_fig_7.html"><a href="13-Supp_fig_7.html#combine-plots"><i class="fa fa-check"></i><b>13.3</b> Combine plots</a></li>
</ul></li>
<li class="part"><span><b>Supplementary tables</b></span></li>
<li class="chapter" data-level="14" data-path="14-Supp_table_2.html"><a href="14-Supp_table_2.html"><i class="fa fa-check"></i><b>14</b> Supp table 2</a>
<ul>
<li class="chapter" data-level="14.1" data-path="14-Supp_table_2.html"><a href="14-Supp_table_2.html#libraries-10"><i class="fa fa-check"></i><b>14.1</b> Libraries</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="14-Supp_table_2.html"><a href="14-Supp_table_2.html#statistical-test"><i class="fa fa-check"></i><b>14.1.1</b> Statistical test</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="15-Supp_table_3.html"><a href="15-Supp_table_3.html"><i class="fa fa-check"></i><b>15</b> Supp table 3</a>
<ul>
<li class="chapter" data-level="15.1" data-path="15-Supp_table_3.html"><a href="15-Supp_table_3.html#libraries-11"><i class="fa fa-check"></i><b>15.1</b> Libraries</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="15-Supp_table_3.html"><a href="15-Supp_table_3.html#paired-wilcox-tests"><i class="fa fa-check"></i><b>15.1.1</b> Paired wilcox tests</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="16-Supp_table_4.html"><a href="16-Supp_table_4.html"><i class="fa fa-check"></i><b>16</b> Supp table 4</a>
<ul>
<li class="chapter" data-level="16.1" data-path="16-Supp_table_4.html"><a href="16-Supp_table_4.html#libraries-12"><i class="fa fa-check"></i><b>16.1</b> Libraries</a></li>
<li class="chapter" data-level="16.2" data-path="16-Supp_table_4.html"><a href="16-Supp_table_4.html#significant-results-table"><i class="fa fa-check"></i><b>16.2</b> Significant results table</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="17-Supp_table_6.html"><a href="17-Supp_table_6.html"><i class="fa fa-check"></i><b>17</b> Supp table 6</a>
<ul>
<li class="chapter" data-level="17.1" data-path="17-Supp_table_6.html"><a href="17-Supp_table_6.html#libraries-13"><i class="fa fa-check"></i><b>17.1</b> libraries</a>
<ul>
<li class="chapter" data-level="17.1.1" data-path="17-Supp_table_6.html"><a href="17-Supp_table_6.html#statisitical-tests"><i class="fa fa-check"></i><b>17.1.1</b> Statisitical tests</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="18-Supp_table_7.html"><a href="18-Supp_table_7.html"><i class="fa fa-check"></i><b>18</b> Supp table 7</a></li>
<li class="chapter" data-level="19" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html"><i class="fa fa-check"></i><b>19</b> Supp table 8</a>
<ul>
<li class="chapter" data-level="19.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#libraries-14"><i class="fa fa-check"></i><b>19.1</b> Libraries</a></li>
<li class="chapter" data-level="19.2" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#preprocess-tables-1"><i class="fa fa-check"></i><b>19.2</b> Preprocess tables</a></li>
<li class="chapter" data-level="19.3" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#no-buffer"><i class="fa fa-check"></i><b>19.3</b> No buffer</a>
<ul>
<li class="chapter" data-level="19.3.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-1"><i class="fa fa-check"></i><b>19.3.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.4" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#rnalater"><i class="fa fa-check"></i><b>19.4</b> RNAlater</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-2"><i class="fa fa-check"></i><b>19.4.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#ethanol"><i class="fa fa-check"></i><b>19.5</b> Ethanol</a>
<ul>
<li class="chapter" data-level="19.5.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-3"><i class="fa fa-check"></i><b>19.5.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.6" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#psp"><i class="fa fa-check"></i><b>19.6</b> PSP</a>
<ul>
<li class="chapter" data-level="19.6.1" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#correlation-4"><i class="fa fa-check"></i><b>19.6.1</b> Correlation</a></li>
</ul></li>
<li class="chapter" data-level="19.7" data-path="19-Supp_table_8.html"><a href="19-Supp_table_8.html#compare-results"><i class="fa fa-check"></i><b>19.7</b> Compare results</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Optimised human stool sample collection for multi-omic microbiota analysis R code</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="fig6" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Figure 6<a href="06-Figure_6.html#fig6" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Correlation of bacterial genera with SCFA metabolites. Spearman’s correlation was carried out on all the samples with matching ASV (≥4,528 reads) and SCFA data. Relative abundance values of SCFAs and genera were used for correlation. Only genera with significant correlations are displayed (adjusted p-value (fdr) &lt;0.05), all correlation values between the SCFAs and these genera are displayed. The y-axis represents the SCFA metabolites. The x-axis represents the genera of bacteria (based on 16S rRNA gene sequences). Red and blue indicate positive and negative correlation, respectively.</p>
<p><img src="figures/Figure%206.jpg" width="100%" style="display: block; margin: auto;" /></p>
<p>Created with assistance of tutorial: <a href="https://microbiome.github.io/tutorials/Heatmap.html" class="uri">https://microbiome.github.io/tutorials/Heatmap.html</a></p>
<div id="libraries-3" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Libraries<a href="06-Figure_6.html#libraries-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="06-Figure_6.html#cb21-1" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb21-2"><a href="06-Figure_6.html#cb21-2" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb21-3"><a href="06-Figure_6.html#cb21-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb21-4"><a href="06-Figure_6.html#cb21-4" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb21-5"><a href="06-Figure_6.html#cb21-5" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb21-6"><a href="06-Figure_6.html#cb21-6" tabindex="-1"></a><span class="fu">library</span>(funrar)</span></code></pre></div>
</div>
<div id="preprocess-tables" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Preprocess tables<a href="06-Figure_6.html#preprocess-tables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="06-Figure_6.html#cb22-1" tabindex="-1"></a><span class="co">#load in phyloseq object</span></span>
<span id="cb22-2"><a href="06-Figure_6.html#cb22-2" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;./data/preprocess_physeq&quot;</span>)</span>
<span id="cb22-3"><a href="06-Figure_6.html#cb22-3" tabindex="-1"></a><span class="co">#subset samples to remove rnalater unwashed samples</span></span>
<span id="cb22-4"><a href="06-Figure_6.html#cb22-4" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> </span>
<span id="cb22-5"><a href="06-Figure_6.html#cb22-5" tabindex="-1"></a>  <span class="fu">subset_samples</span>(physeq, <span class="sc">!</span>(RNAlater_washed_status <span class="sc">==</span> <span class="st">&quot;unwashed&quot;</span>))</span>
<span id="cb22-6"><a href="06-Figure_6.html#cb22-6" tabindex="-1"></a><span class="co">#Subset samples to remove samples with NA lipid info</span></span>
<span id="cb22-7"><a href="06-Figure_6.html#cb22-7" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> <span class="fu">subset_samples</span>(physeq, <span class="sc">!</span>(Formate.MM. <span class="sc">==</span> <span class="st">&quot;NA&quot;</span>))</span>
<span id="cb22-8"><a href="06-Figure_6.html#cb22-8" tabindex="-1"></a><span class="co">#Remove anything that is not bacteria</span></span>
<span id="cb22-9"><a href="06-Figure_6.html#cb22-9" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> <span class="fu">subset_taxa</span>(physeq, Phylum <span class="sc">!=</span> <span class="st">&quot;Bacteria&quot;</span>)</span>
<span id="cb22-10"><a href="06-Figure_6.html#cb22-10" tabindex="-1"></a></span>
<span id="cb22-11"><a href="06-Figure_6.html#cb22-11" tabindex="-1"></a><span class="co">#Clostridium</span></span>
<span id="cb22-12"><a href="06-Figure_6.html#cb22-12" tabindex="-1"></a><span class="co">#Get logical vector to know which rows </span></span>
<span id="cb22-13"><a href="06-Figure_6.html#cb22-13" tabindex="-1"></a>clos_subset_vector <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">replace</span>((<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="sc">==</span> <span class="st">&quot;Clostridium&quot;</span>), </span>
<span id="cb22-14"><a href="06-Figure_6.html#cb22-14" tabindex="-1"></a>        <span class="fu">is.na</span>(<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="sc">==</span> <span class="st">&quot;Clostridium&quot;</span>),</span>
<span id="cb22-15"><a href="06-Figure_6.html#cb22-15" tabindex="-1"></a>        <span class="cn">FALSE</span>))</span>
<span id="cb22-16"><a href="06-Figure_6.html#cb22-16" tabindex="-1"></a><span class="co">#Extract taxa names</span></span>
<span id="cb22-17"><a href="06-Figure_6.html#cb22-17" tabindex="-1"></a>clos_taxa_names <span class="ot">&lt;-</span> <span class="fu">taxa_names</span>(physeq)[clos_subset_vector]</span>
<span id="cb22-18"><a href="06-Figure_6.html#cb22-18" tabindex="-1"></a><span class="co">#Make new genus names for clositriudm</span></span>
<span id="cb22-19"><a href="06-Figure_6.html#cb22-19" tabindex="-1"></a>clos_new_genus_name <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Family&quot;</span>], </span>
<span id="cb22-20"><a href="06-Figure_6.html#cb22-20" tabindex="-1"></a>                             <span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Genus&quot;</span>], </span>
<span id="cb22-21"><a href="06-Figure_6.html#cb22-21" tabindex="-1"></a>                             <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb22-22"><a href="06-Figure_6.html#cb22-22" tabindex="-1"></a><span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Genus&quot;</span>] <span class="ot">&lt;-</span> clos_new_genus_name</span>
<span id="cb22-23"><a href="06-Figure_6.html#cb22-23" tabindex="-1"></a><span class="co">#Ruminococcus</span></span>
<span id="cb22-24"><a href="06-Figure_6.html#cb22-24" tabindex="-1"></a><span class="co">#Remove instances of square brackets</span></span>
<span id="cb22-25"><a href="06-Figure_6.html#cb22-25" tabindex="-1"></a>clos_subset_vector <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">replace</span>((<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="sc">==</span> <span class="st">&quot;[Ruminococcus]&quot;</span>), </span>
<span id="cb22-26"><a href="06-Figure_6.html#cb22-26" tabindex="-1"></a>        <span class="fu">is.na</span>(<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="sc">==</span> <span class="st">&quot;[Ruminococcus]&quot;</span>),</span>
<span id="cb22-27"><a href="06-Figure_6.html#cb22-27" tabindex="-1"></a>        <span class="cn">FALSE</span>))</span>
<span id="cb22-28"><a href="06-Figure_6.html#cb22-28" tabindex="-1"></a><span class="co">#Remove []</span></span>
<span id="cb22-29"><a href="06-Figure_6.html#cb22-29" tabindex="-1"></a>clos_taxa_names <span class="ot">&lt;-</span> <span class="fu">taxa_names</span>(physeq)[clos_subset_vector]</span>
<span id="cb22-30"><a href="06-Figure_6.html#cb22-30" tabindex="-1"></a><span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Genus&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Ruminococcus&quot;</span></span>
<span id="cb22-31"><a href="06-Figure_6.html#cb22-31" tabindex="-1"></a><span class="co">#Get logical vector to know which rows </span></span>
<span id="cb22-32"><a href="06-Figure_6.html#cb22-32" tabindex="-1"></a>clos_subset_vector <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">replace</span>((<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="sc">==</span> <span class="st">&quot;Ruminococcus&quot;</span>), </span>
<span id="cb22-33"><a href="06-Figure_6.html#cb22-33" tabindex="-1"></a>        <span class="fu">is.na</span>(<span class="fu">tax_table</span>(physeq)[,<span class="st">&quot;Genus&quot;</span>] <span class="sc">==</span> <span class="st">&quot;Ruminococcus&quot;</span>),</span>
<span id="cb22-34"><a href="06-Figure_6.html#cb22-34" tabindex="-1"></a>        <span class="cn">FALSE</span>))</span>
<span id="cb22-35"><a href="06-Figure_6.html#cb22-35" tabindex="-1"></a><span class="co">#Extract taxa names</span></span>
<span id="cb22-36"><a href="06-Figure_6.html#cb22-36" tabindex="-1"></a>clos_taxa_names <span class="ot">&lt;-</span> <span class="fu">taxa_names</span>(physeq)[clos_subset_vector]</span>
<span id="cb22-37"><a href="06-Figure_6.html#cb22-37" tabindex="-1"></a><span class="co">#Make new genus names for clositriudm</span></span>
<span id="cb22-38"><a href="06-Figure_6.html#cb22-38" tabindex="-1"></a>clos_new_genus_name <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Family&quot;</span>], </span>
<span id="cb22-39"><a href="06-Figure_6.html#cb22-39" tabindex="-1"></a>                             <span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Genus&quot;</span>], </span>
<span id="cb22-40"><a href="06-Figure_6.html#cb22-40" tabindex="-1"></a>                             <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb22-41"><a href="06-Figure_6.html#cb22-41" tabindex="-1"></a><span class="fu">tax_table</span>(physeq)[clos_taxa_names,<span class="st">&quot;Genus&quot;</span>] <span class="ot">&lt;-</span> clos_new_genus_name</span>
<span id="cb22-42"><a href="06-Figure_6.html#cb22-42" tabindex="-1"></a></span>
<span id="cb22-43"><a href="06-Figure_6.html#cb22-43" tabindex="-1"></a></span>
<span id="cb22-44"><a href="06-Figure_6.html#cb22-44" tabindex="-1"></a><span class="co">#Convert to genus table</span></span>
<span id="cb22-45"><a href="06-Figure_6.html#cb22-45" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> <span class="fu">aggregate_taxa</span>(physeq, <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb22-46"><a href="06-Figure_6.html#cb22-46" tabindex="-1"></a><span class="co">#COnvert to relative abundance</span></span>
<span id="cb22-47"><a href="06-Figure_6.html#cb22-47" tabindex="-1"></a>physeq_relabund <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">transform</span>(physeq, <span class="st">&quot;compositional&quot;</span>)</span>
<span id="cb22-48"><a href="06-Figure_6.html#cb22-48" tabindex="-1"></a><span class="co">#Extract genus table to a df</span></span>
<span id="cb22-49"><a href="06-Figure_6.html#cb22-49" tabindex="-1"></a>genus_df <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">otu_table</span>(physeq_relabund))</span>
<span id="cb22-50"><a href="06-Figure_6.html#cb22-50" tabindex="-1"></a><span class="co">#transpose</span></span>
<span id="cb22-51"><a href="06-Figure_6.html#cb22-51" tabindex="-1"></a>genus_df_t <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(genus_df))</span>
<span id="cb22-52"><a href="06-Figure_6.html#cb22-52" tabindex="-1"></a><span class="co">#log10 the values</span></span>
<span id="cb22-53"><a href="06-Figure_6.html#cb22-53" tabindex="-1"></a><span class="co">#genus_df_t_log10 &lt;- log10(genus_df_t)</span></span>
<span id="cb22-54"><a href="06-Figure_6.html#cb22-54" tabindex="-1"></a></span>
<span id="cb22-55"><a href="06-Figure_6.html#cb22-55" tabindex="-1"></a><span class="co">#extract lipid table</span></span>
<span id="cb22-56"><a href="06-Figure_6.html#cb22-56" tabindex="-1"></a>lipid_mat <span class="ot">&lt;-</span> </span>
<span id="cb22-57"><a href="06-Figure_6.html#cb22-57" tabindex="-1"></a>  <span class="fu">as.matrix</span>(<span class="fu">meta</span>(physeq)[,<span class="fu">grepl</span>(<span class="st">&quot;.MM.&quot;</span>,<span class="fu">colnames</span>(<span class="fu">meta</span>(physeq)))])</span>
<span id="cb22-58"><a href="06-Figure_6.html#cb22-58" tabindex="-1"></a><span class="co">#remove .MM. from colnames</span></span>
<span id="cb22-59"><a href="06-Figure_6.html#cb22-59" tabindex="-1"></a><span class="fu">colnames</span>(lipid_mat) <span class="ot">&lt;-</span> </span>
<span id="cb22-60"><a href="06-Figure_6.html#cb22-60" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;.MM.&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(lipid_mat))</span>
<span id="cb22-61"><a href="06-Figure_6.html#cb22-61" tabindex="-1"></a><span class="co">#Remove &quot;_b&quot; from sample names</span></span>
<span id="cb22-62"><a href="06-Figure_6.html#cb22-62" tabindex="-1"></a><span class="fu">row.names</span>(lipid_mat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;_b&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb22-63"><a href="06-Figure_6.html#cb22-63" tabindex="-1"></a>                             <span class="fu">row.names</span>(lipid_mat))</span>
<span id="cb22-64"><a href="06-Figure_6.html#cb22-64" tabindex="-1"></a><span class="fu">row.names</span>(genus_df_t) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="at">pattern =</span> <span class="st">&quot;_b&quot;</span>, <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb22-65"><a href="06-Figure_6.html#cb22-65" tabindex="-1"></a>                             <span class="fu">row.names</span>(genus_df_t))</span>
<span id="cb22-66"><a href="06-Figure_6.html#cb22-66" tabindex="-1"></a><span class="co">#Convert to relative abundance values</span></span>
<span id="cb22-67"><a href="06-Figure_6.html#cb22-67" tabindex="-1"></a>lipid_mat_relabund <span class="ot">&lt;-</span> <span class="fu">make_relative</span>(lipid_mat)</span>
<span id="cb22-68"><a href="06-Figure_6.html#cb22-68" tabindex="-1"></a></span>
<span id="cb22-69"><a href="06-Figure_6.html#cb22-69" tabindex="-1"></a><span class="co">#Save object</span></span>
<span id="cb22-70"><a href="06-Figure_6.html#cb22-70" tabindex="-1"></a><span class="fu">save</span>(genus_df_t, <span class="at">file =</span> <span class="st">&quot;./data/16s_and_lipid_genus_physeq&quot;</span>)</span>
<span id="cb22-71"><a href="06-Figure_6.html#cb22-71" tabindex="-1"></a><span class="fu">save</span>(lipid_mat_relabund, <span class="at">file =</span> <span class="st">&quot;./data/lipid_relabund_matrix&quot;</span>)</span></code></pre></div>
</div>
<div id="all-buffers" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> All buffers<a href="06-Figure_6.html#all-buffers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="correlation" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Correlation<a href="06-Figure_6.html#correlation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="06-Figure_6.html#cb23-1" tabindex="-1"></a><span class="co">#load in phyloseq object</span></span>
<span id="cb23-2"><a href="06-Figure_6.html#cb23-2" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;./data/16s_and_lipid_genus_physeq&quot;</span>)</span>
<span id="cb23-3"><a href="06-Figure_6.html#cb23-3" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;./data/lipid_relabund_matrix&quot;</span>)</span>
<span id="cb23-4"><a href="06-Figure_6.html#cb23-4" tabindex="-1"></a><span class="co">#Correlation</span></span>
<span id="cb23-5"><a href="06-Figure_6.html#cb23-5" tabindex="-1"></a>correlation.table <span class="ot">&lt;-</span> <span class="fu">associate</span>(</span>
<span id="cb23-6"><a href="06-Figure_6.html#cb23-6" tabindex="-1"></a>  genus_df_t, </span>
<span id="cb23-7"><a href="06-Figure_6.html#cb23-7" tabindex="-1"></a>  lipid_mat_relabund, </span>
<span id="cb23-8"><a href="06-Figure_6.html#cb23-8" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;table&quot;</span>, </span>
<span id="cb23-9"><a href="06-Figure_6.html#cb23-9" tabindex="-1"></a>  <span class="at">p.adj.threshold =</span> <span class="fl">0.05</span>, <span class="at">n.signif =</span> <span class="dv">1</span>)</span>
<span id="cb23-10"><a href="06-Figure_6.html#cb23-10" tabindex="-1"></a><span class="co">#Add column names</span></span>
<span id="cb23-11"><a href="06-Figure_6.html#cb23-11" tabindex="-1"></a><span class="fu">colnames</span>(correlation.table)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Genus&quot;</span>,<span class="st">&quot;Lipid&quot;</span>)</span>
<span id="cb23-12"><a href="06-Figure_6.html#cb23-12" tabindex="-1"></a><span class="co">#Write table with significant hits to file</span></span>
<span id="cb23-13"><a href="06-Figure_6.html#cb23-13" tabindex="-1"></a><span class="fu">write.table</span>(correlation.table[correlation.table<span class="sc">$</span>p.adj <span class="sc">&lt;</span> <span class="fl">0.05</span>,], </span>
<span id="cb23-14"><a href="06-Figure_6.html#cb23-14" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;./data/Genus_lipid_sig_correlation.csv&quot;</span>, </span>
<span id="cb23-15"><a href="06-Figure_6.html#cb23-15" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb23-16"><a href="06-Figure_6.html#cb23-16" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-14286e82776c265e6eb0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-14286e82776c265e6eb0">{"x":{"filter":"none","vertical":false,"data":[["cc_115","Lachnobacterium","Blautia","Erysipelotrichaceae_Clostridium","Clostridiaceae_Clostridium","Roseburia","Cloacibacillus","Dehalobacterium","Coprococcus","Enterococcus","Oribacterium","Anaerostipes","Alistipes","Gemella","Dehalobacterium","Dialister","Anaerostipes","Barnesiella","Haemophilus","Clostridiaceae_Clostridium","Cloacibacillus","Butyricicoccus","Clostridiaceae_Clostridium","Oxalobacter","Bacteroides","Veillonella","Granulicatella","Oxalobacter","Dehalobacterium","rc4-4","SMB53","Peptostreptococcaceae_Clostridium","Unknown","Anaerostipes","RFN20","Cloacibacillus","Porphyromonas","SMB53","Veillonella","Oxalobacter","Lachnospira","Roseburia","Faecalibacterium","Pelomonas","Pelomonas","Desulfovibrio","Prevotella","Anaerotruncus","cc_115","Caldicoprobacter","Subdoligranulum","Pelomonas","Veillonella","Lachnobacterium","Pelomonas","Pelomonas","Pelomonas","Paraprevotella","Bacteroides","Lachnospiraceae_Ruminococcus","Aggregatibacter","Coprococcus","[Eubacterium]","Sutterella","Bilophila","Butyricimonas","cc_115","Dorea","Caldicoprobacter","Butyricicoccus","Bifidobacterium","Campylobacter","[Eubacterium]","Slackia","Desulfovibrio","Sutterella","Peptostreptococcaceae_Clostridium","Haemophilus","Peptoniphilus","Blautia","Peptoniphilus","Oscillospira","Ruminococcaceae_Ruminococcus","Christensenella","Bacteroides","Adlercreutzia","Slackia","Adlercreutzia","cc_115","Bacteroides","Streptococcus","Alistipes","Anaerofustis","Faecalibacterium","Unknown","SMB53","Lachnospiraceae_Ruminococcus","Coprobacillus","Anaerofustis","Lachnospira","cc_115","WAL_1855D"],["isobutyrate","Valerate","Formate","Acetate","Propionate","isobutyrate","isobutyrate","isovalerate","Valerate","Succinate","Succinate","Butyrate","Valerate","Succinate","isobutyrate","Propionate","Propionate","Valerate","isobutyrate","Butyrate","isovalerate","Formate","isovalerate","Butyrate","Propionate","isobutyrate","Lactate","isovalerate","Butyrate","Propionate","isobutyrate","Propionate","isobutyrate","isovalerate","Propionate","Butyrate","isobutyrate","Propionate","Valerate","isobutyrate","isobutyrate","Acetate","isobutyrate","Butyrate","isovalerate","isovalerate","Valerate","isobutyrate","Butyrate","Propionate","Butyrate","Propionate","Propionate","isobutyrate","Acetate","isobutyrate","Valerate","Valerate","Valerate","Succinate","isobutyrate","Propionate","Acetate","isobutyrate","Acetate","Valerate","isovalerate","isovalerate","Acetate","isobutyrate","Valerate","Butyrate","isobutyrate","Valerate","Butyrate","Acetate","Succinate","Propionate","isobutyrate","Acetate","Valerate","Acetate","Propionate","isovalerate","Butyrate","Valerate","Butyrate","isobutyrate","Acetate","isovalerate","Valerate","isovalerate","Butyrate","isovalerate","Acetate","Valerate","Acetate","Propionate","isovalerate","isovalerate","Valerate","isobutyrate"],[0.529575655685674,-0.524795174396469,0.49976807884914,0.504009052325944,0.510037276247776,-0.502565341310417,0.495743563243352,0.49885103889586,-0.508450385220988,0.508203649186155,0.497155743769064,-0.491163130192116,0.474627994763297,0.475059932934884,0.46180108276612,-0.45804594764356,-0.453154069152811,-0.453023330272324,-0.439344005236153,0.441062190818004,0.439173159562476,0.436033487679888,-0.436550234938458,-0.434958323962674,0.430428749417889,-0.430931721992761,0.42723422683442,0.410922573875732,-0.409590317587261,-0.406030354850776,-0.405375233772326,-0.401375431719758,0.400183287545352,0.393293978540636,-0.391490241162257,-0.391393822642274,0.384989966935399,-0.380311744367924,-0.380130010704513,0.379334595059773,-0.378464058052162,-0.370360094735779,-0.366183033636212,-0.364170556716247,0.364171917356333,0.365120736303007,-0.363408800502445,0.361076649714622,-0.35918683061589,-0.358287588957615,-0.356638464541475,-0.355479654687534,-0.349490389866392,-0.349866084945042,-0.346847028050992,-0.346847028050992,-0.346847028050992,-0.344108080737168,0.338178426964348,0.338491199369636,-0.336666385474381,-0.335938773144024,0.33307088243187,-0.332990173619374,-0.331729072785199,-0.331723717003863,0.329077949034804,-0.327280627549107,0.325593120455463,-0.32389520407402,-0.321507238262197,-0.318565789575215,0.317902924330119,0.314025432174708,-0.3125411584898,-0.309112246841495,0.305721410723963,-0.30177393015518,0.299627472583532,-0.297161237108078,0.294938160981312,0.294184033046632,-0.293378126638562,0.293443462255328,0.292532241358603,-0.292107931707189,-0.291037483246108,-0.290086096614837,0.289183958640996,-0.288904666638426,-0.289442929489677,-0.288180426377302,-0.286312479406858,-0.285753848182815,0.284636250513763,-0.284913436338399,0.283974293074231,-0.281815175929214,0.280706221479251,-0.280315029096989,0.28056687236653,0.279255411118137],[2.48997176721905e-05,2.48997176721905e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,2.95945029562985e-05,3.59106931870906e-05,8.19495431220571e-05,8.19495431220571e-05,0.000157944654843592,0.00018208933767969,0.000212619488079855,0.000212619488079855,0.000378261940275596,0.000378261940275596,0.000378261940275596,0.000405760328674468,0.000405760328674468,0.000410764992519469,0.000476701898883764,0.000476701898883764,0.000538418675871065,0.00114364417946407,0.00117567636678998,0.00133888447268752,0.00133888447268752,0.00155940913148981,0.0015967815046376,0.0021144768575648,0.00217307039707305,0.00217307039707305,0.00280034858716101,0.00327572628537219,0.00327572628537219,0.0033041854586743,0.00334535957041408,0.00458911408757134,0.00532315264340005,0.00540120672323907,0.00540120672323907,0.00540120672323907,0.00545217372951494,0.00586547370880012,0.00619791249550744,0.00629584773351045,0.00659041621085359,0.00676683784106532,0.0082354183347217,0.0082354183347217,0.00863882273609497,0.00863882273609497,0.00863882273609497,0.00942643955818212,0.0113924653040156,0.0113924653040156,0.0118541171041712,0.0119817383269241,0.0129391844628451,0.0129391844628451,0.0131420678826903,0.0131420678826903,0.0142526069357568,0.0149837281270204,0.0156879139282474,0.0164291194615691,0.0176273497156027,0.0192732966773337,0.0194540452701411,0.0219486158576254,0.0227873977576027,0.0252703868118788,0.0279551837603047,0.031462764352322,0.0333351330253065,0.0356748921781294,0.0378611246191415,0.0383179981269201,0.0383856246627027,0.0383856246627027,0.0389747715400677,0.0390469702254448,0.0399365314285506,0.0406928292092783,0.0408488524892915,0.0408488524892915,0.0408488524892915,0.0413374787724867,0.0433605070615839,0.0436542886911772,0.0442586843115999,0.0442586843115999,0.0447118562687136,0.0473079468932856,0.0480656291131991,0.0480656291131991,0.0480656291131991,0.049160466043896]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Genus<\/th>\n      <th>Lipid<\/th>\n      <th>Correlation<\/th>\n      <th>p.adj<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":50,"columnDefs":[{"className":"dt-right","targets":[2,3]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="06-Figure_6.html#cb24-1" tabindex="-1"></a><span class="co">#Create wide table of sig correlations</span></span>
<span id="cb24-2"><a href="06-Figure_6.html#cb24-2" tabindex="-1"></a>long_df <span class="ot">&lt;-</span> correlation.table[correlation.table<span class="sc">$</span>p.adj <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb24-3"><a href="06-Figure_6.html#cb24-3" tabindex="-1"></a><span class="co">#Remove p.adj value</span></span>
<span id="cb24-4"><a href="06-Figure_6.html#cb24-4" tabindex="-1"></a>long_df <span class="ot">&lt;-</span> long_df[,<span class="sc">-</span><span class="dv">4</span>]</span>
<span id="cb24-5"><a href="06-Figure_6.html#cb24-5" tabindex="-1"></a><span class="co">#long to wide</span></span>
<span id="cb24-6"><a href="06-Figure_6.html#cb24-6" tabindex="-1"></a>wide_df <span class="ot">&lt;-</span> <span class="fu">dcast</span>(long_df, Genus <span class="sc">~</span> Lipid, <span class="at">value.var =</span> <span class="st">&quot;Correlation&quot;</span>)</span>
<span id="cb24-7"><a href="06-Figure_6.html#cb24-7" tabindex="-1"></a><span class="co">#Write to file</span></span>
<span id="cb24-8"><a href="06-Figure_6.html#cb24-8" tabindex="-1"></a><span class="fu">write.table</span>(wide_df, </span>
<span id="cb24-9"><a href="06-Figure_6.html#cb24-9" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;./data/Genus_lipid_sig_correlation_wide.csv&quot;</span>, </span>
<span id="cb24-10"><a href="06-Figure_6.html#cb24-10" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb24-11"><a href="06-Figure_6.html#cb24-11" tabindex="-1"></a>            <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="heatmap" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Heatmap<a href="06-Figure_6.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="06-Figure_6.html#cb25-1" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>(<span class="dv">20</span>))</span>
<span id="cb25-2"><a href="06-Figure_6.html#cb25-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> correlation.table</span>
<span id="cb25-3"><a href="06-Figure_6.html#cb25-3" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(Genus, Lipid, <span class="at">group=</span>Lipid)) </span>
<span id="cb25-4"><a href="06-Figure_6.html#cb25-4" tabindex="-1"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Correlation)) </span>
<span id="cb25-5"><a href="06-Figure_6.html#cb25-5" tabindex="-1"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(df<span class="sc">$</span>Correlation, <span class="dv">1</span>)), <span class="at">size =</span> <span class="dv">5</span>) </span>
<span id="cb25-6"><a href="06-Figure_6.html#cb25-6" tabindex="-1"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> </span>
<span id="cb25-7"><a href="06-Figure_6.html#cb25-7" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="st">&quot;Correlation&quot;</span>,</span>
<span id="cb25-8"><a href="06-Figure_6.html#cb25-8" tabindex="-1"></a>                       <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">to =</span> <span class="dv">1</span>,  <span class="at">by =</span> <span class="fl">0.25</span>),</span>
<span id="cb25-9"><a href="06-Figure_6.html#cb25-9" tabindex="-1"></a>                       <span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb25-10"><a href="06-Figure_6.html#cb25-10" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb25-11"><a href="06-Figure_6.html#cb25-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb25-12"><a href="06-Figure_6.html#cb25-12" tabindex="-1"></a>        <span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb25-13"><a href="06-Figure_6.html#cb25-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;&quot;</span>) <span class="sc">+</span> </span>
<span id="cb25-14"><a href="06-Figure_6.html#cb25-14" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb25-15"><a href="06-Figure_6.html#cb25-15" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> p,</span>
<span id="cb25-16"><a href="06-Figure_6.html#cb25-16" tabindex="-1"></a>       <span class="at">filename =</span> <span class="st">&quot;./figures/Genus_lipid_corr_heatmap.png&quot;</span>,</span>
<span id="cb25-17"><a href="06-Figure_6.html#cb25-17" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">&quot;png&quot;</span>, <span class="at">units =</span> <span class="st">&quot;mm&quot;</span>, </span>
<span id="cb25-18"><a href="06-Figure_6.html#cb25-18" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">400</span>, <span class="at">width =</span> <span class="dv">300</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code></pre></div>
<p><img src="figures/Genus_lipid_corr_heatmap.png" width="100%" style="display: block; margin: auto;" /></p>

</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="05-Figure_5.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="07-Supp_fig_1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
